src/utils/src/deviceinfo.cpp

```diff
@@ -16,49 +16,49 @@
  */
 
 #include "deviceinfo.h"
+#include <QFile>
+#include <QCoreApplication>
+
+const char* CONFIG_FILE = "/etc/machine.conf";
+
+DeviceInfo::DeviceInfo()
+{
+    QCoreApplication::setOrganizationName("AsteroidOS");
+    QCoreApplication::setOrganizationDomain("asteroidos.org");
+    QCoreApplication::setApplicationName("Device Info");
+    m_settings = new QSettings(CONFIG_FILE, QSettings::IniFormat);
```

Comment:
If you want a heap-allocated QObject to get garbage collected automatically you should provide a parent. Otherwise this might cause a memory leak. I suspect we have tons of those in the codebase already because I wasn't particularly careful with those in the past. Maybe you can also just have m_settings be a simple QSettings member.

Reply:
Now a simple member.